### PR TITLE
26 update 2nd handshake signature validation

### PR DESCRIPTION
**Pull request by DarksightKellar**
This PR updates the service to use unencrypted shard _double hashes_ instead of single hashes when signing off on their assigned shard, and adds `maxRewrapInterval` to the list of data to validate from embalmer.

#### NOTES
 - Added some structure to the Arweave sign-off signature that is sent to the browser, and a try-catch so that the browser should receive a response even in an exception.
 
 - The webapp would now have to JSON decode this string, first check if it has an `error` field, _before_ attempting to access the `signature` field, which will hold the signature of the archaeologist, or an empty string to indicate the archaeologist did NOT sign off.
 
 - If the response has an error field, it will be an object `{ code: number, message: string }`. `code` can be checked to formulate a user-friendly message to show the user, or perform other correction logic. 
 Find error codes in `utils/error-codes.ts`
 
 - The stream protocol to use in `confirmArweaveTransaction` in the webapp should be updated from `/validate-arweave` to `/arweave-signoff`

 - The stream protocol to handle in the webapp for receiving the node's public key should be updated from `/env-config` to `/public-key`

### CODE REVIEW

File: service/src/models/archaeologist.ts [hunk 1]
```diff
@@ -124,24 +125,41 @@ export class Archaeologist {
...
-                streamToBrowser(signature);
-              } else {
-                streamToBrowser('0');
+            for await (const data of source) {
+              try {
+                const jsonData: {
+                  arweaveTxId: string,
+                  unencryptedShardDoubleHash: string,
+                  maxRewrapInterval: number,
+                } = JSON.parse(new TextDecoder().decode(data));
```
Comment (sethhrbek) [inline service/src/models/archaeologist.ts:138]: If you wanted to get around having to re-assign variables from this json object (or call `jsonData.x`), could define an interface outside of this function somewhere and use it, like:

```
interface SarcoDataFromEmbalmerToValidate {
  arweaveTxId: string;
  unencryptedShardDoubleHash: string;
  maxRewrapInterval;
}
...
const {
    arweaveTxId,
    unencryptedShardDoubleHash,
    maxRewrapInterval
  }: SarcoDataFromEmbalmerToValidate = JSON.parse(new TextDecoder().decode(data));
```

File: service/src/models/archaeologist.ts [hunk 2]
```diff
@@ -124,24 +125,41 @@ export class Archaeologist {
...
-            for await (const data of source) {
-              const jsonData = JSON.parse(new TextDecoder().decode(data));
-
-              const txId = jsonData.arweaveTxId;
-              const unencryptedShardHash = jsonData.unencryptedShardHash;
-
-              const isValidShard = await fetchAndValidateArweaveShard(txId, unencryptedShardHash, this.web3Interface.encryptionWallet.publicKey);
+            const rejectTransaction = () => streamToBrowser(JSON.stringify({ signature: '' }))
+            const signOff = (signature: string) => streamToBrowser(JSON.stringify({ signature }));
+            const emitError = (error: any) => streamToBrowser(JSON.stringify({ error }));
```
[the remark this answers is not in the document]
Comment (DarksightKellar) [reply service/src/models/archaeologist.ts:130]: Ah good points on this and your related comment above. I'll build out an initial concept, from where we can co-ordinate with the webapp

File: service/src/models/archaeologist.ts (hunk 1 above)
Comment (DarksightKellar) [reply service/src/models/archaeologist.ts:138]: Oh sweet, gotta love destructuring. Thanks for the reminder!

**Comment by DarksightKellar** [comment]
@sethhrbek -- Added commits to address your comments. 

I also updated `/env-config` to `/public-key`.

**Pull request merged**